torch/lib/THD/base/Cuda.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Cuda.hpp"
+
+THCState** _THDCudaState;
```

Comment:
That seems janky. Can't you just initialize THD after constructing the THCState*?